### ISSUE TITLE
make nework JSON serialiable

### ISSUE THREAD

**Issue by ttinoco** [ttinoco/PFNET.py#7]
right now it can be pickled but that does not seem to make it json serializable automatically

**Comment by ttinoco** [comment]
jsonpickle does the job

**Comment by awig** [comment]
this one? https://jsonpickle.github.io/

what are the advantages of having the network json object serializable?

**Comment by ttinoco** [comment]
yes, that one. pfnet networks are already json serializable but the serialization/deserialization has to be done manually (net.json_string and using a ParserJSON for the other direction). A benefit of json is readability. Django uses JSON for saving objects and was throwing errors when encountering custom data types that included or were networks. For pickle one can add specific methods to a class to make it serializable and routines like pickle.dumps and pickle.loads work automatically. For pfnet networks, the json string and ParserJSON are used under the hood to make them pickleable. I was not able to find a similar way to make json.loads/json.dumps work automatically. I found that one needs to create and pass it a custom encoder that knows how to json-serialize custom types. This path did not seem promising. I found jsonpickle, which seemed to work well like pickle. I ran into error using jsonpickle with custom objects that contained networks and ended up concatenating jsonpickle with pickle to go from complex python object to bytes to string and viceversa.
